### PR TITLE
Address Err in char_rnn tutorial issue

### LINKED ISSUE #193: Err in char_rnn tutorial
@apaszke    
This tutorial "[char_rnn_classification](http://pytorch.org/tutorials/intermediate/char_rnn_classification_tutorial.html)" has a bug in the forward part of this code:   

```python    
import torch.nn as nn
from torch.autograd import Variable

class RNN(nn.Module):
    def __init__(self, input_size, hidden_size, output_size):
        super(RNN, self).__init__()

        self.hidden_size = hidden_size

        self.i2h = nn.Linear(input_size + hidden_size, hidden_size)
        self.i2o = nn.Linear(input_size + hidden_size, output_size)
        self.softmax = nn.LogSoftmax(dim=1)

    def forward(self, input, hidden):
        combined = torch.cat((input, hidden), 1)
        hidden = self.i2h(combined)
        output = self.i2o(combined)
        output = self.softmax(output)
        return output, hidden

    def initHidden(self):
        return Variable(torch.zeros(1, self.hidden_size))

n_hidden = 128
rnn = RNN(n_letters, n_hidden, n_categories)
```   

The RNN formula is:   
image

Which is implemented by these lines:    
```python    
        combined = torch.cat((input, hidden), 1)
        hidden = self.i2h(combined)
```    

However, these lines:    
```python    
        output = self.i2o(combined)   
        output = self.softmax(output)
```   
Are trying to project into the classification space. However, the self.i2o operates on the combined output instead of the ht output.    

This implementation uses the wrong formula:    
image

But the correct formula is:      
image

     
Which can be implemented as:   
```python    
    def forward(self, input, hidden):
        combined = torch.cat((input, hidden), 1)
        hidden = self.i2h(combined)
        output = self.i2o(hidden) # this line changed   (bc hidden = ht. combined = ht-1)
        output = self.softmax(output)
        return output, hidden
```    

Basically, the current implementation does this:    
image

But it really should do this:    
image

### PR DESCRIPTION
**Pull request by mikebrow**
Fixes #193

## Description
- Address issue described Err in char_rnn tutorial;
- Also did a small amount of text only editing / cleanup missing links to related tutorials, correct a subtitle

## Checklist

- [x] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [x] Only one issue is addressed in this pull request
- [x] Labels from the issue that this PR is fixing are added to this pull request
- [x] No unnessessary issues are included into this pull request.

### CODE REVIEW

**Comment by mikebrow** [comment]
rebased off main, odd the clone was only a couple hours old :) 

I have internal administrator working on the fb cla, hopefully will check out correctly in the next hour..?

**Review by kit1980 — APPROVED** [review]

File: intermediate_source/char_rnn_classification_tutorial.py [hunk 1]
```diff
@@ -4,11 +4,14 @@
...
 **Author**: `Sean Robertson <https://github.com/spro>`_
 
-We will be building and training a basic character-level RNN to classify
-words. This tutorial, along with the following two, show how to do
-preprocess data for NLP modeling "from scratch", in particular not using
-many of the convenience functions of `torchtext`, so you can see how
-preprocessing for NLP modeling works at a low level.
+We will be building and training a basic character-level Recurrent Neural
+Network (RNN) to classify words. This tutorial, along with two other
+Natural Language Procesing (NLP) "from scratch" tutorials
```
Comment (svekars) [inline intermediate_source/char_rnn_classification_tutorial.py:9]:
```suggestion
Natural Language Processing (NLP) "from scratch" tutorials
```

Comment (mikebrow) [reply intermediate_source/char_rnn_classification_tutorial.py]: thx :)

**Review by mikebrow — COMMENTED** [review]
SGTM
